src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs

```diff
@@ -120,6 +120,39 @@ public void RemoveShouldPreserveIntermediaryReferences(string content)
             ObjectModelHelpers.AssertItems(new string[0], itemsForI2);
         }
 
+        [Fact]
```

Comment:
I'd add tests for both types of item transforms:
- item functions `@(foo->function())`
- string templatey thing: `@(foo->'%(filename)')`